### PR TITLE
feat: set up light/dark mode css variables

### PR DESCRIPTION
**Pull request by tisidro**
### Description:

Set up light and dark-mode CSS variables to set the colors used throughout the site.

### Spec:

Designs: [Designs](DESIGN_URL)

See Story: [FSA22V2-246](https://sparkbox.atlassian.net/browse/FSA22V2-246?atlOrigin=eyJpIjoiMDhmYzQ2YzBkYThjNDE4OGE3ZDI1ZjFiZDAzNjY5MmEiLCJwIjoiaiJ9)

### Validation:

Added color map, created variables, updated variables throughout the site. Please note: The image at the top of the home page will appear black when dark mode is enabled. That will be addressed in a future card and should not block approval of this card. Thanks!

- [x] This PR has code changes, and our linters still pass.
- [x] This PR affects production code, so it was browser tested (see below).
- [x] This PR has new code, so new tests were added or updated, and they pass.

#### To Validate:

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, etc).
2. Pull down all related branches.
3. In terminal: `npm run dev`.
4. Check [localhost:3000](http://localhost:3000/)for changes.
5. Run `npm run lint`.
6. Run axe Devtools on affected pages.
7. Confirm all tests pass.

---

#### Browser Testing

In these browsers, behavior & design closely match original specifications. A user is able to access all content and functionality, including the usability of required assistive devices, such as keyboard and screenreader.

**macOS**

- [ ] Safari (last 2 major versions)
- [ ] Chrome (last 6 months)
- [ ] Firefox (last 6 months)

**Windows**

- [ ] Chrome (last 6 months)
- [ ] Firefox (last 6 months)
- [ ] Edge (last 6 months)

**Mobile**

- [ ] Safari (last 2 major versions)
- [ ] Chrome on Android (last 6 months)
- [ ] Firefox on Android (last 6 months)

### CODE REVIEW

File: src/styles/components/_component-grid.scss [hunk 1]
```diff
@@ -11,14 +11,11 @@ $breakpoint-tablet: 820px;
...
-<<<<<<< HEAD
     margin: 0;
-    background-color: white;
+    background-color: var(--card-background-color);
     text-align: center;
     align-items: flex-start;
-=======
-    background-color: var(--background-color);
->>>>>>> a4258c0 (feat: add dark mode for text, logo, and borders)
+
```
Comment (rkelbel) [inline src/styles/components/_component-grid.scss:18]: Your linter is mad about an extra line here, you can delete one of the lines of 18 or 19 and it should be all 😄 again.

**Review by rkelbel — COMMENTED** [review]
This is coming along well Terri, let me know if you need any other additional questions on what we discussed in pairing. 🙂

File: src/styles/components/_home.scss [hunk 2]
```diff
@@ -5,9 +5,12 @@ $breakpoint-tablet: 820px;
   grid-template-columns: 1fr;
   gap: 1rem;
   width: 100%;
+  background-color: var(--background-color);
+  color: var(--font-color);
 
   &-top__top-image {
     width: 100%;
+    background-color: var(--background-color);
   }
```
Comment (bsmrdel101) [inline src/styles/components/_home.scss:15]: If what you're saying is true, that Dustin doesn't want this image to appear in dark mode, then you can just remove it like this.
```suggestion
  @media (prefers-color-scheme: dark) {
    &-top__top-image {
      display: none;
    }
  }
```

**Review by bsmrdel101 — COMMENTED** [review]
Good job! I'm able to switch my preference to dark mode and the website will change. I just left a comment regarding the home image.

File: src/styles/components/_home.scss (hunk 2 above)
Comment (tisidro) [reply src/styles/components/_home.scss:15]: Actually, what he meant is that styling the image to appear is outside the scope of this card. If it appears blacked-out on this card, that is fine.

File: src/styles/components/_definition.scss [hunk 3]
```diff
@@ -18,7 +20,7 @@ $breakpoint-tablet: 820px;
 
   @media only screen and (min-width: $breakpoint-tablet) {
     background:
-      linear-gradient(to right, white 50%, rgba(255 255 255 / 0%)),
+      linear-gradient(to right, var(--background-color) 50%, transparent),
```
Comment (bsmrdel101) [inline src/styles/components/_definition.scss:23]: Screen Shot 2022-10-31 at 9 38 45 AM

If you want to get rid of this visual bug, you can just switch the gradient color to black on dark mode.

File: src/styles/components/_footer.scss [hunk 4]
```diff
@@ -22,6 +22,7 @@
 
   // Flex items of content wrapper
   &__item {
+    color: var(--foreground-color);
```
Comment (bsmrdel101) [inline src/styles/components/_footer.scss:25]: **Current site:**
Screen Shot 2022-10-31 at 9 44 12 AM
Screen Shot 2022-10-31 at 9 44 05 AM

The dark mode theme in figma shows that this should have a dark background with white text and icons.

**Figma:**
Screen Shot 2022-10-31 at 9 45 31 AM

**Review by bsmrdel101 — CHANGES_REQUESTED** [review]
It's great that you opened this PR again! I left a couple of fixes to visual bugs while in dark mode.

Another thing I noticed on light mode is that the sparkbox logo background color changes too black, which makes in no longer visible.
Screen Shot 2022-10-31 at 9 50 40 AM

File: src/styles/components/_definition.scss (hunk 3 above)
Comment (tisidro) [reply src/styles/components/_definition.scss:23]: Thanks for catching that Bennett! :) As for the Sparkbox logo--I'm not sure what is supposed to happen w/ that. I need to ask Jelly as in the design the logo is actually black on dark mode.

File: src/styles/components/_component-card.scss [hunk 5]
```diff
@@ -6,13 +6,14 @@
 
   .component-card {
     margin: 0;
-    border: 2px solid black;
+    border: 2px solid var(--foreground-color);
```
Comment (bsmrdel101) [inline src/styles/components/_component-card.scss:9]: Good job moving color values into their own variable!

File: src/styles/components/_definition.scss [hunk 6]
```diff
@@ -18,8 +20,8 @@ $breakpoint-tablet: 820px;
 
   @media only screen and (min-width: $breakpoint-tablet) {
     background:
-      linear-gradient(to right, white 50%, rgba(255 255 255 / 0%)),
-      url("../../../public/definition_text_light.svg");
+      linear-gradient(to right, var(--background-color) 50%, transparent),
```
Comment (bsmrdel101) [inline src/styles/components/_definition.scss:23]: Nice fix

File: src/styles/tools/_functions.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+@function color($color) {
+  @return map.get($colors, $color);
```
Comment (bsmrdel101) [inline src/styles/tools/_functions.scss:2]: This is wild.

**Review by bsmrdel101 — CHANGES_REQUESTED** [review]
Screen Shot 2022-10-31 at 2 50 55 PM
The text color here doesn't match the figma design, otherwise everything looks great!

**Pull request merged**